app/frontend/src/main.js

```diff
@@ -196,21 +196,60 @@ function loadKeycloak(config) {
         authStore.ready = true;
         typeof options.onReady === 'function' && options.onReady();
       };
+
+      let updateTokenInterval;
+      // The expired token is also used for general update token failures
+      // i.e., network disconnection
+      let expiredTokenInterval;
+
+      function updateToken(seconds) {
+        try {
+          keycloak
+            .updateToken(seconds)
+            .success((refreshed) => {
+              if (refreshed) {
+                if (expiredTokenInterval) clearInterval(expiredTokenInterval);
+              }
+            })
+            .error(() => {
+              // Failed to refresh token
+              if (!expiredTokenInterval)
+                expiredTokenInterval = setInterval(() => {
+                  updateToken(60);
+                }, 10000);
+              // If we're failing to update the token, we will clear the token
+              keycloak.clearToken();
+            })
+            .catch(() => {
+              // If we're failing to update the token, we will clear the token
+              keycloak.clearToken();
+            });
+        } catch (error) {
+          // If we're failing to update the token, we will clear the token
+          keycloak.clearToken();
```

Comment:
We don't do anything with the errors because so far, it seems like the errors that are caught or thrown are just undefined or a boolean value.